scene/main/viewport.cpp

```diff
@@ -2355,6 +2341,20 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 	}
 
 	if (mm.is_null() && mb.is_null() && p_event->is_action_type()) {
+		const bool resize_allowed = !DisplayServer::get_singleton()->window_get_flag(DisplayServer::WINDOW_FLAG_RESIZE_DISABLED);
+
+		if (p_event->is_pressed() && p_event->is_action_pressed("ui_toggle_fullscreen") && resize_allowed) {
```

Comment:
```suggestion
		if (p_event->is_pressed() && p_event->is_action_pressed("ui_toggle_fullscreen")) {
```

See previous discussion. Let's not create artificial restrictions. If a person has `window_resizable == false` and they don't need fullscreen, then they can simply remove all events from the `ui_toggle_fullscreen` action.

Reply:
If the window is not resizable, then we should *not* allow fullscreen to not break expectations.

If you really want to make a non-resizable window fullscreen, you can use a window manager that allows ignoring geometry restrictions like KWin.

Reply:
I agree with @Calinou here, if resizing a window isn't allowed, it shouldn't be resized, whatever keys you press.